alerts/openshift-container-storage-operator/StorageAutoScalingFailed.md

```diff
@@ -0,0 +1,76 @@
+# StorageAutoScalingFailed
+
+## Meaning
+
+During automatic storage scaling, failures may occur due to various factors. In
+such instances, the `StorageAutoScalingFailed` alert is fired.
+
+## Impact
+
+The storage cannot be scaled at the moment.
+
+## Diagnosis
+
+There are multiple stages where the storage configurations might not have matched
+the requested storage parameters.
+
+1. OSD pod is in `Pending` state:
+    If it is in `Pending` state with the following logs, there is a high chance
+    it is due to the exhaustion of the resources.
+    ```bash
+    Warning  FailedScheduling  3m53s  default-scheduler  0/6 nodes are available: 1 Insufficient cpu, 2 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/6 nodes are available: 1 Insufficient cpu, 5 Preemption is not helpful for scheduling.
+    ```
+
+    Describe the pod in `Pending` state and check for the events.
+    ```bash
+    oc describe pod <osd_pod> -n <namespace>
+    ```
+
+    If there are no errors, retrieve the pvc of the pod in `Pending` state.
+    ```bash
+    pvcName=$(oc get pod <osd_pod> -n <namespace> -o jsonpath='{.spec.volumes[*].persistentVolumeClaim.claimName}')
+    oc get pvc $pvcName -n <namespace> -o yaml
+    ```
+    Check for the health and events of the PVC for detailed information.
+
+2. PVC is in `Pending` state:
+
+    ```bash
+    oc get pvc <pvc_name> -n <namespace> -o yaml
+    ```
+
+    Inspect the events of the pvc. Additionally, retrieve the pv of the pvc if
+    it is still in `pending` state.
+    ```bash
+    oc get pvc <pvc_name> -n <namespace> -o jsonpath='{.spec.volumeName}'
+    ```
+
+3. PV is in `Pending` state:
```

Comment:
While the PVC might be in pending state, I don't know if I've ever seen a PV in pending state. @parth-gr Have you ever seen that?

Reply:
Not both at the same time